overlord/snapstate/backend/setup_test.go

```diff
@@ -144,7 +145,14 @@ func (s *setupSuite) TestSetupDoUndoInstance(c *C) {
 }
 
 func (s *setupSuite) TestSetupDoUndoKernelUboot(c *C) {
-	loader := boottest.NewMockBootloader("mock", c.MkDir())
+	// setup uboot config, so we can get uboot bootloader
+	mockGadgetDir := c.MkDir()
```

Comment:
This bit needs a bit of refactoring I think, ideally we would see if we can make this work with the mockbootloader, i will try to look at this in my morning

Reply:
@mvo5 depends what we want to test, with mocbootloader extraction test is not giving us any value

Reply:
Indeed, looking at this now.